frontend/src/js/components/PageViewer/FindInput.tsx

```diff
@@ -1,42 +1,46 @@
-import _ from "lodash";
+import _, { uniq } from 'lodash';
 import React, {
   FC,
   KeyboardEventHandler,
   useCallback,
-  useEffect,
+  useEffect, useMemo,
   useState,
-} from "react";
+} from 'react';
 import DownIcon from "react-icons/lib/md/arrow-downward";
 import UpIcon from "react-icons/lib/md/arrow-upward";
 import styles from "./FindInput.module.css";
+import { HighlightForSearchNavigation } from './model';
+import { Loader } from 'semantic-ui-react';
 
 type FindInputProps = {
   value: string;
   setValue: (v: string) => void;
   performFind: (query: string) => Promise<void>;
+  isPending: boolean;
   jumpToNextFindHit: () => void;
   jumpToPreviousFindHit: () => void;
-  hits: number[];
-  lastPageHit: number;
+  highlights: HighlightForSearchNavigation[];
+  focusedFindHighlightIndex: number | null;
 };
 
 // The backend will only return 500 pages of hits.
 // If we get that many then we need to inform the user that there could be missing values.
 // In the future we can make a paging system for find search hits.
-const MAX_HITS = 500;
+const MAX_PAGE_HITS = 500;
 
 export const FindInput: FC<FindInputProps> = ({
   value,
   setValue,
   jumpToNextFindHit,
   jumpToPreviousFindHit,
   performFind,
-  hits,
-  lastPageHit,
+  isPending,
+  highlights,
+  focusedFindHighlightIndex,
 }) => {
   const [showWarning, setShowWarning] = useState(false);
 
-  const debouncedPerformSearch = useCallback(_.debounce(performFind, 300), [
```

Comment:
Any reason this was changed to `useMemo`? The behavior should be identical but `useCallback` is more idiomatic, though personally I'm not very fussed about idiomatic details like this.

Reply:
Technically with the `useCallback` approach `_.debounce` gets called again each render because of the evaluation order: it has to evaluate the inner function call to `debounce` before it passes result to `useCallback`. Though `useCallback` won't use the result unless the deps have changed. 

Whereas with `useMemo` you can ensure that if you're calling a higher-order function you only re-call it when necessary.

This is a micro-optimisation though really as `debounce` is not an expensive operation. What do you prefer?